builder/builder-next/controller.go

```diff
@@ -446,15 +446,16 @@ func getGCPolicy(conf config.BuilderConfig, root string) ([]client.PruneInfo, er
 		} else {
 			gcPolicy = make([]client.PruneInfo, len(conf.GC.Policy))
 			for i, p := range conf.GC.Policy {
-				var b int64
+				var b = defaultKeepStorage
 				if p.KeepStorage != "" {
 					b, err = units.RAMInBytes(p.KeepStorage)
 					if err != nil {
 						return nil, errors.Wrapf(err, "failed to parse keepStorage")
 					}
-				}
-				if b == 0 {
-					b = defaultKeepStorage
```

Comment:
Looks like this was always there from the start, but we probably didn't run into it because of the error above; https://github.com/moby/moby/commit/4a776d0ca76c4bdf4399aef8c102361d6c2819eb

- https://github.com/moby/moby/pull/37846